app/Domain/Clients/Controllers/ShowClient.php

```diff
@@ -52,38 +54,95 @@ public function init(
             }
         }
 
+
+        public function get($params): Response
+        {
+            Auth::authOrRedirect([Roles::$owner, Roles::$admin, Roles::$manager], true);
+
+            $id = '';
+
+            if (isset($params['id']) === true) {
+                $id = (int) ($params['id']);
+            }
+
+            $row = $this->clientService->get($id);
+
+            if ($row === false) {
+                $this->tpl->display('errors.error404');
+
+                return $this->tpl->display('errors.error403');
+            }
+
+            $clientValues = (object) [
+                'id' => $row->id,
+                'name' => $row->name,
+                'street' => $row->street,
+                'zip' => $row->zip,
+                'city' => $row->city,
+                'state' => $row->state,
+                'country' => $row->country,
+                'phone' => $row->phone,
+                'internet' => $row->internet,
+                'email' => $row->email,
+            ];
+
+
+            if (empty($row) === false && Auth::userIsAtLeast(Roles::$admin)) {
+                $file = app()->make(FileRepository::class);
+                $project = app()->make(ProjectRepository::class);
+
+                if (session('userdata.role') == 'admin') {
+                    $this->tpl->assign('admin', true);
+                }
+
+                $this->tpl->assign('userClients', $this->clientService->getUserClients($id));
+                $this->tpl->assign('comments', $this->commentService->getComments('client', $id));
+                $this->tpl->assign('imgExtensions', ['jpg', 'jpeg', 'png', 'gif', 'psd', 'bmp', 'tif', 'thm', 'yuv']);
+                $this->tpl->assign('client', $clientValues);
+                $this->tpl->assign('users', app()->make(UserRepository::class));
```

Comment:
Let's pass in the user service here and make the call in the controller. Haven't seen the template yet but I assume there is a repo method call in the template which sould probably not be in there.